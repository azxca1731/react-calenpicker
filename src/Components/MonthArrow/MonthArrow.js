import React from "react";
import PropTypes from "prop-types";
import { CssConnector, PropsConnector } from "../../Containers/Provider";
import dark from "./MonthArrow.style.dark";
import light from "./MonthArrow.style.light";

class MonthArrow extends React.Component {
  constructor(props) {
    super(props);
    this.style = props.theme == "light" ? light : dark;
  }

  render() {
    const { type, onClick, cssObject } = this.props;
    return (
      <div
        className={this.style.MonthArrow}
        onClick={onClick}
        style={cssObject}
      >
        {type === "left" ? "❮" : "❯"}
      </div>
    );
  }
}

MonthArrow.propTypes = {
  type: PropTypes.oneOf(["left", "right"]).isRequired,
  onClick: PropTypes.func.isRequired,
  cssObject: PropTypes.object,
  theme: PropTypes.string
};

export default PropsConnector(({ state }) => ({
  theme: state.theme
}))(
  CssConnector(({ state }) => ({
    cssObject: state.MonthArrowCssObject
  }))(MonthArrow)
);
