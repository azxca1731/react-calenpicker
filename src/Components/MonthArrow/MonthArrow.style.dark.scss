@import "../../Styles/theme/dark";
@import "./MonthArrow.style";
