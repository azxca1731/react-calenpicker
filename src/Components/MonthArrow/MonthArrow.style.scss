.MonthArrow {
  padding: 25px;
}
