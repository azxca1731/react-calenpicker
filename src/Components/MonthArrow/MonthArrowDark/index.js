import MonthArrowDark from "./MonthArrowDark";

export default MonthArrowDark;
