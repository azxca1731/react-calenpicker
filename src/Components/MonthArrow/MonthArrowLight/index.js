import MonthArrowLight from "./MonthArrowLight";

export default MonthArrowLight;
