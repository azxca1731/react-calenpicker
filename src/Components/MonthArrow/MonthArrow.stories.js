import React from "react";
import { ThemeProvider } from "styled-components";
import { storiesOf } from "@storybook/react";
import { withInfo } from "@storybook/addon-info";

import light from "Styles/theme/light";
import dark from "Styles/theme/dark";
import MonthArrow from "Components/MonthArrow";

storiesOf("Components", module)
  .addDecorator(withInfo)
  .addDecorator(story => <ThemeProvider theme={light}>{story()}</ThemeProvider>)
  .add("MonthArrow light", () => <MonthArrow />);

storiesOf("Components", module)
  .addDecorator(withInfo)
  .addDecorator(story => <ThemeProvider theme={dark}>{story()}</ThemeProvider>)
  .add("MonthArrow Dark", () => <MonthArrow />);
