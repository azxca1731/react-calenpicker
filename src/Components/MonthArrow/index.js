import MonthArrow from "./MonthArrow";

export default MonthArrow;
