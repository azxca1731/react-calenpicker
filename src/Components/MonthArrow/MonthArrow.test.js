import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

import MonthArrow from "./index";

describe("MonthArrow", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<MonthArrow />);
    expect(wrapper).toHaveLength(1);
  });

  it("renders with no props", () => {
    const wrapper = mount(<MonthArrow />);
    expect(wrapper.text()).toContain("❮");
  });

  it("renders props for type to right", () => {
    const wrapper = mount(<MonthArrow type="right" />);
    expect(wrapper.text()).toContain("❯");
  });

  it("renders props for type to none", () => {
    const wrapper = mount(<MonthArrow type="none" />);
    expect(wrapper.text()).toContain(" ");
  });

  it("renders with snapshot", () => {
    const wrapper = mount(<MonthArrow />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("render custom css render well with snapshot", () => {
    const customCssObject = { color: "red" };
    const wrapper = shallow(<MonthArrow cssObject={customCssObject} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(toJson(wrapper).props.style).toBe(customCssObject);
  });
});
