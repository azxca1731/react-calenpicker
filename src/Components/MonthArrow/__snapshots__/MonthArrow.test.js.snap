// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthArrow render custom css render well with snapshot 1`] = `
<MonthArrow
  cssObject={
    Object {
      "color": "red",
    }
  }
  onClick={[Function]}
  type="left"
>
  <styled.div
    onClick={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "srDMb",
            "rules": Array [
              "
  padding: 25px;
  cursor: pointer;
  color: ",
              [Function],
              ";
  font-size: 130%;
  background-color: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
      style={
        Object {
          "color": "red",
        }
      }
    >
      <div
        className="sc-bdVaJa srDMb"
        onClick={[Function]}
        style={
          Object {
            "color": "red",
          }
        }
      >
        ❮
      </div>
    </StyledComponent>
  </styled.div>
</MonthArrow>
`;

exports[`MonthArrow renders with snapshot 1`] = `
<MonthArrow
  cssObject={Object {}}
  onClick={[Function]}
  type="left"
>
  <styled.div
    onClick={[Function]}
    style={Object {}}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "srDMb",
            "rules": Array [
              "
  padding: 25px;
  cursor: pointer;
  color: ",
              [Function],
              ";
  font-size: 130%;
  background-color: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
      style={Object {}}
    >
      <div
        className="sc-bdVaJa srDMb"
        onClick={[Function]}
        style={Object {}}
      >
        ❮
      </div>
    </StyledComponent>
  </styled.div>
</MonthArrow>
`;
